subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/serialization/codecs/MethodCodec.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.instantexecution.serialization.codecs
+
+import org.gradle.instantexecution.serialization.Codec
+import org.gradle.instantexecution.serialization.ReadContext
+import org.gradle.instantexecution.serialization.WriteContext
+import org.gradle.instantexecution.serialization.readClass
+import org.gradle.instantexecution.serialization.writeClass
+import java.lang.reflect.Method
+
+
+internal
+object MethodCodec : Codec<Method> {
+
+    override fun WriteContext.encode(value: Method) {
+        writeClass(value.declaringClass)
+        writeString(value.name)
+        write(value.parameterTypes)
```

Comment:
:x: It would be more performant and maybe a little nicer to read with a `writeClasses` method assembled using the `writeCollection` combinator (`writeCollection(Arrays.asList(types)) { writeClass(it) }`)